gstd/src/msg/async.rs

```diff
@@ -48,6 +55,9 @@ where
                 return Poll::Ready(Err(ContractError::ExitCode(exit_code)));
             }
 
+            // Remove lock after waking.
```

Comment:
Shouldn't this be also executed if logic in lines 54-56 (things about about returning error reply) triggered?